request_test.go

```diff
@@ -504,82 +504,212 @@ func TestRequest_URLConcatenate(t *testing.T) {
 
 	reporter := NewAssertReporter(t)
 
-	config1 := Config{
-		BaseURL:  "",
-		Client:   client,
-		Reporter: reporter,
-	}
-
-	config2 := Config{
-		BaseURL:  "http://example.com",
-		Client:   client,
-		Reporter: reporter,
+	cases := []struct {
+		name      string
+		baseURL   string
+		method    string
+		path      string
+		pathArgs  []interface{}
+		setupFunc func(request *Request)
+		fullURL   string
```

Comment:
Let's rename fullURL to expectedURL or something like this, to make it clear that it's not input parameter, but expected output.